data_model/src/predicate.rs

```diff
@@ -2,54 +2,155 @@
 
 #[cfg(not(feature = "std"))]
 use alloc::vec;
+use core::{fmt::Display, ops::Not};
 
 use super::*;
 use crate::{IdBox, Name, Value};
 
+/// Struct representing sequence of non-zero length
+#[derive(Debug, Clone, Serialize, Deserialize, Encode, Decode, IntoSchema)]
+pub struct NonEmpty<T> {
+    head: Box<T>,
```

Comment:
Why do we want `Box<T>` and not just `T`?

Reply:
`GenericPredicateBox` becomes self-referential otherwise